source/mainnet/smart-contracts/tutorials/counter/counter-contract.rst

```diff
@@ -25,215 +25,211 @@ Once you have set up the tools needed you are ready to create your smart contrac
+
+.. code-block:: console
+
+    $concordium-client contract init <MODULE-HASH> \
+        --sender <YOUR-TESTNET-ADDRESS> \
+        --energy 30000 \
+        --contract counter \
+        --grpc-ip grpc.testnet.concordium.com \
+        --grpc-port 20000 \
+        --secure
+
+Be sure to note down the contract index returned by this command. You'll need the index in the next section to invoke functions for the contract instance.
+
+Congratulations if you made it this far! You have now successfully deployed and initialized a simple smart contract.
 
 Interact with the contract
 ==========================
 
 View function
 -------------
 
-First, check the initial state of the contract.
+First, check the initial state of the contract. Use this command if you're hosting your own node.
 
 .. code-block:: console
 
-    $concordium-client contract invoke <YOUR-CONTRACT-INSTANCE> --entrypoint view --schema dist/schema.bin --grpc-port 20001
+    $concordium-client contract invoke <CONTRACT-INSTANCE-INDEX> \
+        --entrypoint view \
+        --grpc-port 20001
 
-Since you just initialized the contract it is 0.
+Or, use this command to use the Concordium testnet node:
+
+.. code-block:: console
 
-.. image:: images/invoke.png
-    :width: 100%
+    $concordium-client contract invoke <CONTRACT-INSTANCE-INDEX> \
+        --entrypoint view \
+        --grpc-ip grpc.testnet.concordium.com \
+        --grpc-port 20000 \
+        --secure
 
-Increment function
+Since you just initialized the contract, you should see that the return value is 0.
+
+Update function
 ------------------
 
-Create a JSON file that holds your operator that will be given as input to the function and run the command below. Basically, you are saying to the contract instance “with this transaction I will update your state from the increment entrypoint” which is your function name with this parameter.
+In order to call a function that takes input, like our update function, we'll need to create a JSON file that represents the input to the function. Since our input in this simple example is just a number, a simple text file with a number will do, since this is also valid JSON. We can quickly make this file with this command:
 
 .. code-block:: console
 
-    $concordium-client contract update <YOUR-CONTRACT-INSTANCE> --entrypoint increment --parameter-json <PATH-TO-JSON> --schema dist/smart-contract-multi/schema.bin --sender <YOUR-ADDRESS> --energy 6000 --grpc-port 20001
-
-Start by testing with your conditions. First, try another account other than the owner of the contract since you want that only the owner can call this function.
+    echo 42 > input.json
 
-.. image:: images/owner-error.png
-    :width: 100%
+Now we can invoke the update function with that input by using a contract update transaction. From the point of this transaction in the blockchain, your smart contract will have its new value. If you have your own node, you can invoke the update function like so:
```

Comment:
I guess I'm not using the terminology correctly or maybe not understanding completely.

I mean that by invoking the update function, some kind of transaction or something will be written on the blockchain. Is that not correct? I.e. some kind of transaction will say "this function was invoked with these parameters" and from that point forward in the blockchain, the state will be mutated according to what the function did.

Do you understand what I mean? Maybe you could suggest a better wording?

Reply:
Maybe:
```
Now we can invoke the update function with that input by using a contract update transaction. This will mutate the smart contract state and store the new value. If you have your own node, you can invoke the update function like so:
```